### PR TITLE
Simplified parser implementation

### PR DESCRIPTION
**Pull request by andyg0808**
WIP PR to allow commenting on the implementation.

Auto-dev-blocked as a draft PR.

### CODE REVIEW

**Review by andyg0808 — COMMENTED** [review]
Here's a first-pass of feedback

File: src/markup_to_json.test.js [hunk 1]
```diff
@@ -1,49 +1,52 @@
...
 describe("markup_to_jsonJSON", () => {
-  test('Markup converts to expected JSON', async () => {
-    const testMarkup = fs.readFileSync("./test/993333.markup", "utf8");
-    const testJSON = await convertMarkupToJSON(
-      testMarkup,
-      "./test/993333.source.jpg",
-      "./literally.anywhere"
-    );
+  test("Markup converts to expected JSON", async () => {
+    TestCases.forEach(async (testcase) => {
```
[the remark this answers is not in the document]
Comment (djmetzle) [reply src/markup_to_json.test.js:18]: That's cool. I like that.

File: src/new_markup_to_json.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
+const { Int } = require("./utils");
+const GM = require("gm");
+
+const pos = (x, y) => ({ x: Int(x), y: Int(y) });
+const dim = (x, y) => ({ width: Int(x), height: Int(y) });
+const color = '(black|red|orange|yellow|green|blue|indigo|violet)'
+
+const MarkupParser = {
+  crop: {
+    regex: new RegExp(`crop,(\\d+)x(\\d+),(\\d+)x(\\d+)`),
```
[the remark this answers is not in the document]
Comment (djmetzle) [reply src/new_markup_to_json.js:10]: Oh, we could. I just converted all four for symmetry.

File: src/new_markup_to_json.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    t: (m) => ({
+      block: "crop",
+      instruction: {
+        from: pos(m[1], m[2]),
+        size: dim(m[3], m[4]),
+      },
+    }),
+  },
+  circle: {
+    regex: new RegExp(`circle,(\\d+)x(\\d+),(\\d+),${color}`),
```
[the remark this answers is not in the document]
Comment (djmetzle) [reply src/new_markup_to_json.js:20]: Its not _too_ bad...

File: src/new_markup_to_json.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        color: m[6],
+      },
+    }),
+  },
+};
+
+function convertMarkupToJSON(markup, infile, outfile, stroke) {
+  return new Promise((resolve) => {
+    var json = {};
+    GMGetSize(infile).then((size) => {
```
[the remark this answers is not in the document]
Comment (djmetzle) [reply src/new_markup_to_json.js:60]: Now that we're on Node ~8~ 16 we can `utils.promisify` for clarity and correctness i think?
https://nodejs.org/docs/latest-v8.x/api/util.html#util_util_promisify_original

File: src/new_markup_to_json.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        json.instructions.strokeWidth = stroke;
+      }
+
+      resolve(json);
+    });
+  });
+}
+
+function parseInstruction(instruction) {
+  //console.info(`instruction: ${JSON.stringify(instruction)}`);
```
[the remark this answers is not in the document]
Comment (djmetzle) [reply src/new_markup_to_json.js:99]: My bad. Debug info snuck in, huh?

**Comment by andyg0808** [comment]
Closing in favor of #68.

**Pull request closed without merging**